### PR TITLE
Add a problem with bounds and constraints

### PR DESCRIPTION
**Pull request by tmigot**
I noticed in https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/pull/50 that the coverage is decreasing because we don't have a problem with both bounds and constraints in the problem list.

https://github.com/JuliaSmoothOptimizers/OptimizationProblems.jl/blob/main/src/ADNLPProblems/hs13.jl

### CODE REVIEW

**Review by dpo — COMMENTED** [review]
HS13 is degenerate. Many solvers won't be able to solve it. What will it be used for?

**Comment by tmigot** [comment]
We usually consider the problems from this package to test the NLPModel API, so we never try to solve them.
They are used in the following four packages only: `ADNLPModels`, `NLPModelsJuMP`, `CUTEst`, `AmplNLReader`.

File: src/NLPModelsTest.jl [hunk 1]
```diff
@@ -6,7 +6,7 @@ using LinearAlgebra, SparseArrays, Test
 using NLPModels, NLPModelsModifiers
 
 const nlp_problems =
-  ["BROWNDEN", "HS5", "HS6", "HS10", "HS11", "HS14", "LINCON", "LINSV", "MGH01Feas"]
+  ["BROWNDEN", "HS5", "HS6", "HS10", "HS13", "HS11", "HS14", "LINCON", "LINSV", "MGH01Feas"]
```
Comment (abelsiqueira) [inline src/NLPModelsTest.jl:9]:
```suggestion
  ["BROWNDEN", "HS5", "HS6", "HS10", "HS11", "HS13", "HS14", "LINCON", "LINSV", "MGH01Feas"]
```

File: src/nlp/problems/hs13.jl [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+"""
+    nlp = HS13()
+
+## Problem 13 in the Hock-Schittkowski suite
+
+```math
+\\begin{aligned}
+\\min \\quad & (x_1 - 2)^2 + x_2^2 \\\\
+\\text{s. to} \\quad & 0 \\geq (1 - x_1)^3 - x_2
```
Comment (abelsiqueira) [inline src/nlp/problems/hs13.jl:11]:
```suggestion
\\text{s. to} \\quad & (1 - x_1)^3 - x_2 \\geq 0
```

File: src/nlp/problems/hs13.jl [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+mutable struct HS13{T, S} <: AbstractNLPModel{T, S}
+  meta::NLPModelMeta{T, S}
+  counters::Counters
+end
+
+function HS13(::Type{T}) where {T}
+  meta = NLPModelMeta{T, Vector{T}}(
+    2,
+    ncon = 1,
+    x0 = T[-2; 2],
```
Comment (abelsiqueira) [inline src/nlp/problems/hs13.jl:26]:
```suggestion
    x0 = T[-2; -2],
```

File: src/nlp/problems/hs13.jl [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+end
+
+function NLPModels.hess_structure!(nlp::HS13, rows::AbstractVector{Int}, cols::AbstractVector{Int})
+  @lencheck 3 rows cols
+  rows[1] = 1
+  rows[2] = 2
+  rows[3] = 2
+  cols[1] = 1
+  cols[2] = 1
+  cols[3] = 2
```
Comment (abelsiqueira) [inline src/nlp/problems/hs13.jl:56]: You don't need `nnzh = 3`. Mixed derivatives are 0. `hess_coord` should be fixed as well.

File: src/nlp/problems/hs13.jl [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  increment!(nlp, :neval_jac)
+  vals .= T[-3 * (1 - x[1])^2; -1]
+  return vals
+end
+
+function NLPModels.jprod!(nlp::HS13, x::AbstractVector, v::AbstractVector, Jv::AbstractVector)
+  @lencheck 2 x v
+  @lencheck 1 Jv
+  increment!(nlp, :neval_jprod)
+  Jv .= [(-3 * (1 - x[1])^2) * v[1] - v[2]]
```
Comment (abelsiqueira) [inline src/nlp/problems/hs13.jl:142]:
```suggestion
  Jv .= (-3 * (1 - x[1])^2) * v[1] - v[2]
```

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks for the addition, it is good to have more coverage. I made some suggestions.

**Review by abelsiqueira — APPROVED** [review]

**Comment by abelsiqueira** [comment]
Thanks for the update

**Pull request merged**